lib/models/request.js

```diff
@@ -0,0 +1,46 @@
+var mongoose = require('mongoose'),
+Schema = require('mongoose').Schema,
+validators = require('mongoose-validators');
+
+var capitalize = function(val){
+  return val.toUpperCase();
+},
+lowercase = function(val){
+  return val.toLowerCase();
+};
+var Header = new Schema({
+  key : {
+    type : String
+  },
+  value : {
+    type : String
+  }
+}),
+Request = new Schema({
+  url: {
+    type: String,
+    required: true,
+    validate : validators.isURL()
+  },
+  method: {
+    type: String,
+    required: true,
+    "enum" : ['GET', 'POST', 'PUT', 'PATCH', 'DELETE', 'HEAD', 'OPTIONS'],
+    set: capitalize
+  },
+  // TODO: Should this be just part of the headers? 
+  contentType : {
```

Comment:
On the TODO, I'd agree and probably not treat these as a special case, since `Content-Type` is just a header like you say

totally small thing, but the set for `toLowerCase()` is probably unnecessary, since header field names are case-insensitive according to the RFC - arguably it looks tidier tho!